packages/muelu/src/Misc/MueLu_SegregatedAFactory_def.hpp

```diff
@@ -51,96 +51,119 @@
 
 #include "MueLu_SegregatedAFactory_decl.hpp"
 
-#include "MueLu_FactoryManager.hpp"
 #include "MueLu_Level.hpp"
 #include "MueLu_Monitor.hpp"
+#include "MueLu_Utilities_decl.hpp"
 
 namespace MueLu {
 
 template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
 RCP<const ParameterList> SegregatedAFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::GetValidParameterList() const {
   RCP<ParameterList> validParamList = rcp(new ParameterList());
 
-#define SET_VALID_ENTRY(name) validParamList->setEntry(name, MasterList::getEntry(name))
-#undef SET_VALID_ENTRY
+  validParamList->set<RCP<const FactoryBase>>("A", Teuchos::null, "Generating factory of the matrix A to be filtered");
+  validParamList->set<std::string>("droppingScheme", "vague", "Strategy to drop entries from matrix A based on the input of some map(s) [blockmap, map-pair]");
 
-  validParamList->set<RCP<const FactoryBase>>("A", Teuchos::null, "Generating factory of the matrix A used for filtering");
-
-  validParamList->set<std::string>("map: name", "", "Name of map (Xpetra::Map) provided by user containing the special DOFs.");
-  validParamList->set<std::string>("map: factory", "", "Name of generating factory for 'map: name'");
+  validParamList->set<RCP<const FactoryBase>>("dropMap1", Teuchos::null, "Generating factory for dropMap1");  ////
+  validParamList->set<RCP<const FactoryBase>>("dropMap2", Teuchos::null, "Generating factory for dropMap2'");
 
   return validParamList;
-}
+}  // GetValidParameterList
 
 template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-void SegregatedAFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::DeclareInput(Level& currentLevel) const {
+void SegregatedAFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::DeclareInput(Level &currentLevel) const {
+  const ParameterList &pL = GetParameterList();
+
+  TEUCHOS_TEST_FOR_EXCEPTION(pL.get<RCP<const FactoryBase>>("A") == Teuchos::null, Exceptions::InvalidArgument,
+                             "Please specify a generating factory for the matrix \"A\" to be filtered.")
+  TEUCHOS_TEST_FOR_EXCEPTION(pL.get<std::string>("droppingScheme") == "vague", Exceptions::InvalidArgument,
+                             "Input map type not selected. Please select one of the available strategies.")
```

Comment:
Would it make more sense to use one of the valid algorithms as default value?

Reply:
I don't necessarily see a reason why one of the schemes should be considered as default. Both have application cases and should be used deliberately. I can change this of course, although I don't really see a reason why.

Reply:
Fine by me.